src/sentry/utils/snuba.py

```diff
@@ -1159,14 +1165,14 @@ def nest_groups(data, groups, aggregate_cols):
             return {c: data[0][c] for c in aggregate_cols} if data else None
     else:
         g, rest = groups[0], groups[1:]
-        inter = {}
+        inter: dict[Any, Any] = {}
         for d in data:
             inter.setdefault(d[g], []).append(d)
         return {k: nest_groups(v, rest, aggregate_cols) for k, v in inter.items()}
 
 
-def resolve_column(dataset) -> Callable[[str], str]:
-    def _resolve_column(col: str) -> str:
+def resolve_column(dataset) -> Callable:
+    def _resolve_column(col: Any) -> Any:
```

Comment:
annotating as `(Any) -> Any` doesn't really help the type checker -- it's better to leave these annotations off so we know it needs to be typed later